File: 231230/그 계절, 그 날/that-season-that-day.java
import java.io.*;
import java.util.*;

public class Main {
    public static class Seanson {
        public static String chooseSeaonFrom(int month) {
            if (month > 11 || month < 3)
                return "Winter";
            
            if (month > 9)
                return "Fall";
            
            if (month > 6)
                return" Summer";
            
            if (month > 3)
                return "Spring";

            return "-1";
        }
    }

    public static boolean isLeapYear(int year) {
        return year % 4 == 0 && (year % 100 != 0 || year % 400 == 0);
    }

    public static boolean validateInput(int y, int m, int d) {
        boolean isValidYear = (y >= 1 && y <= 3000);
        boolean isValidMonth = (m >= 1 && m <= 12);
        boolean isValidDate = (d >= 1 && d <= 31);

        if (m == 2 && d > 28) { // 29
            return isLeapYear(y);
        }

        // 31 months check
        Set<Integer> month31 = new HashSet<>();
        month31.add(1);
        month31.add(3);
        month31.add(5);
        month31.add(7);
        month31.add(8);
        month31.add(10);
        month31.add(12);
        if (d == 31) {
            if (!month31.contains(m))
                return false;
        }
        
        return isValidYear && isValidMonth && isValidDate;
    }

    public static void main(String[] args) throws IOException {
        // input
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));

        String[] input = br.readLine().split(" ");
        
        int year = Integer.parseInt(input[0]);
        int month = Integer.parseInt(input[1]);
        int date = Integer.parseInt(input[2]);

        // validate input & leapyear
        if (!validateInput(year, month, date)) {
            System.out.println("-1");
            return;
        }
        
        // print Seanson
        String seanson = Seanson.chooseSeaonFrom(year);
        
        // print out
        System.out.println(seanson);
    }
}

File: 231230/그 계절, 그 날/that-season-that-day.py
def isYoon(Y):
    leap_Y = False   # 윤년인지 아닌지 체크할 변수로 기본 평년으로 체크
    if Y % 4 == 0:               # 4로 나누어떨어지는지 확인
        if Y % 100 == 0:         # 100으로 나누어떨어지는지 확인
            if Y % 400 == 0:     # 400으로 나누어떨어지는지 확인
                leap_Y = True    # 4, 100, 400으로 나누어떨어지면 윤년
        else:
            leap_Y = True # 4로 나누어떨어지고, 100으로 나누어떨어지지 않으면 윤년

    return leap_Y
def noDay(Y,M,D):
    if (M==4 or M==6 or M==9 or M==11) and D>30:
        return True
    elif M==2 and isYoon(Y)==False and D>28:
        
        return True
    elif M==2 and isYoon(Y)==True and D>29:
        
        return True
    else:
        return False
def solution(Y,M,D):
    if noDay(Y,M,D)==True: #존재하지않으면 return -1
        return -1
    elif 3<=M<=5:
        return "Spring"
    elif 6<=M<=8:
        return "Summer"
    elif 9<=M<=11:
        return "Fall"
    elif M==12 or 1<=M<=2:
        return "Winter"

#입력
Y, M, D= map(int, input().split())
print(solution(Y,M,D))